fds/services/git_service.py

```diff
@@ -24,12 +24,17 @@ def init(self) -> str:
     def status(self) -> Any:
         return execute_command(["git", "status"], capture_output=False)
 
-    def add(self, add_argument: str) -> Any:
+    def add(self, add_argument: str, skipped: List[str]) -> Any:
         git_output = check_git_ignore(add_argument)
         if convert_bytes_to_string(git_output.stdout) != '':
             return
         # This will take care of adding everything in the argument to add including the .dvc files inside it
-        execute_command(["git", "add", add_argument])
+        git_add_command = ["git", "add", add_argument]
+        # Ignore the skipped files if any
+        for skipped_file in skipped:
+            # git add . :!path/to/file1 :!path/to/file2 :!path/to/folder1/* Will ignore the files to be added
+            git_add_command.append(f':!{skipped_file}')
```

Comment:
Seems preferable to use the `:exclude(skipped_file)` syntax to avoid trouble.
Also, you many need to escape potential illegal characters (in this case parentheses) in the path.